### PR TITLE
Add Dev cluster 

### PR DESCRIPTION
**Pull request by hazim1093**
Add Dev cluster a.k.a worker_dev

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -15,9 +15,13 @@ VM_TYPE=hvm
 # For route53.tf
 PRIVATE_DOMAIN=$(CLUSTER_NAME).local
 
+# Development cluster variables, required by worker_dev module
+DEV_APP_FROM_PORT:= 8080
+DEV_APP_TO_PORT:= 8081
```
Comment (rasheedamir) [inline Makefile:27]: @hazim1093 which port is this? user can deploy 1 to M applications to the DEV cluster

File: modules/gocd/conf/cruise-config.xml [hunk 2]
```diff
@@ -141,6 +141,45 @@
...
+            <tasks>
+              <exec command="/bin/bash" workingdir="#{SOURCE_DIR}">
+                <arg>-c</arg>
+                <arg>git config --global user.email "you@example.com"
+                     git config --global user.name "Your Name"</arg>
+                <runif status="passed" />
+              </exec>
+              <exec command="/bin/bash" workingdir="#{SOURCE_DIR}">
+                <arg>-c</arg>
+                <arg>if [ ! -d "/app/app-dev-unit" ]; then sudo mkdir -p /app/app-dev-unit; sudo chown go:go /app/app-dev-unit;fi; \
```
Comment (rasheedamir) [inline modules/gocd/conf/cruise-config.xml]: @hazim1093 i think we shld rename this repo "app-dev-unit"; why have the word "dev"... this is generic

File: resources/policies/worker_policy.json [hunk 3]
```diff
@@ -18,7 +18,7 @@
         "s3:Get*"
       ],
       "Resource": [
-        "arn:aws:s3:::AWS-ACCOUNT-CLUSTER-NAME-cloudinit/etcd/initial-cluster",
+        "arn:aws:s3:::AWS-ACCOUNT-CLUSTER-NAME-cloudinit/CLUSTER-NAME_etcd/initial-cluster",
```
Comment (rasheedamir) [inline resources/policies/worker_policy.json]: @hazim1093 just thinking shouldn't v add the environment name as well?

File: Makefile [hunk 4]
```diff
@@ -15,9 +15,13 @@ VM_TYPE=hvm
...
+
 # For gen-vpc-subnet-modules-tf.sh
 # Add all modules for which <module-name>-subnet.tf needs to be created
-VPC_SUBNET_MODULES=etcd,admiral,docker_registry,worker,elb,rds,aurora_db
+VPC_SUBNET_MODULES=etcd,admiral,docker_registry,worker,elb,rds,worker_dev,aurora_db
```
Comment (rasheedamir) [inline Makefile]: @hazim1093 i think you need to look into this list; what is difference between "worker" & "worker_dev"? also I think we should update "etcd" to "etcd_dev"

File: modules/gocd/conf/cruise-config.xml [hunk 5]
```diff
@@ -141,6 +141,45 @@
...
+                <arg>cd /app/app-dev-unit; sudo mv application.service application-#{APP_NAME}.service</arg>
+                <runif status="passed" />
+              </exec>
+              <exec command="/bin/bash" workingdir="#{SOURCE_DIR}">
+                <arg>-c</arg>
+                <arg>host_ip=$(ip route show | awk '/default/ {print $3}');
+                     fleet_endpoint=$(curl -s -L http://${host_ip}:4001/v2/keys/stakater/dev/ip | jq '.node.value' | sed s/\"//g);
+                     cd /app/app-dev-unit;
+                     fleetctl --etcd-key-prefix=/stakater/dev/ --endpoint=http://${fleet_endpoint}:4001 destroy application-#{APP_NAME}.service
+                     fleetctl --etcd-key-prefix=/stakater/dev/ --endpoint=http://${fleet_endpoint}:4001 start application-#{APP_NAME}.service</arg>
```
Comment (rasheedamir) [inline modules/gocd/conf/cruise-config.xml]: @hazim1093 just wondering on which port the application will be accessible? shouldn't we setup a DNS entry to make the app easily accessible? and include version number in the DNS name as well...

File: Makefile (hunk 1 above)
Comment (hazim1093) [inline Makefile:27]: @rasheedamir so they can open the required ports like
DEV_APP_FROM_PORT:= 1
DEV_APP_TO_PORT:= M

**Pull request merged**